File: SCRUM/ej4/C/greedy_biblioteca_patrones/server/src/main/java/com/ejemplo/biblioteca/web/dto/LibroDTO.java
package com.ejemplo.biblioteca.web.dto;

import com.ejemplo.biblioteca.domain.TipoLibro;

public record LibroDTO(
        Long id,
        String titulo,
        String fecha,
        String genero,
        Integer paginas,
        TipoLibro tipo,
        Double pesoGramos,
        Double tamanoMb,
        AutorDTO autor,
        PersonaSummaryDTO persona
) {
}

File: SCRUM/ej4/C/greedy_biblioteca_patrones/client/src/dto/LibroDTO.ts
import type { AutorDTO } from './AutorDTO';
import type { PersonaDTO } from './PersonaDTO';

export type TipoLibro = 'FISICO' | 'DIGITAL';

export interface LibroDTO {
  id?: number;
  titulo: string;
  fecha: string;
  genero: string;
  paginas: number;
  tipo: TipoLibro;
  pesoGramos?: number | null;
  tamanoMb?: number | null;
  autor: AutorDTO;
  persona: Pick<PersonaDTO, 'id' | 'nombre' | 'apellido'>;
  autorId?: number;
  personaId?: number;
}
